server/app/api/plugin_management.py

```
@@ -0,0 +1,105 @@
+    return request.app.state.plugin_manager
+
+
+@router.get("/plugins")
+def list_plugins(pm=Depends(_pm)):
+    all_names = pm.discover()
+    loaded = list(pm.loaded.keys())
+    return {"available": all_names, "loaded": loaded}
+
+
+@router.post("/plugins/{name}/enable")
+def enable_plugin(name: str, pm=Depends(_pm)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)